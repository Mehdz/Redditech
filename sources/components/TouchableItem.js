import React from 'react';
import {TouchableOpacity} from 'react-native';
import {List} from 'react-native-paper';
import PropTypes from 'prop-types';

const TouchableItem = props => {
    return (
        <TouchableOpacity onPress={props.onPress}>
            <List.Item
                title={props.title}
                left={() => (
                    <List.Icon color={props.color} icon={props.leftIcon} />
                )}
                right={() => (
                    <List.Icon color={props.color} icon={props.rightIcon} />
                )}
            />
        </TouchableOpacity>
    );
};

TouchableItem.propTypes = {
    title: PropTypes.string.isRequired,
    color: PropTypes.string,
    rightIcon: PropTypes.string,
    leftIcon: PropTypes.string,
    onPress: PropTypes.func,
};

export default TouchableItem;
