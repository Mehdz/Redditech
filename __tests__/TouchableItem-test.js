// __tests__/TouchableItem-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import TouchableItem from '../sources/components/Loading';

test('renders correctly', () => {
  const tree = renderer.create(<TouchableItem />).toJSON();
  expect(tree).toMatchSnapshot();
});