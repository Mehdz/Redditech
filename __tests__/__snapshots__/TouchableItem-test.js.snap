// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityRole="progressbar"
  accessibilityState={
    Object {
      "busy": true,
    }
  }
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "flex": 1,
        "marginTop": 50,
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "height": 45,
        "opacity": 1,
        "width": 45,
      }
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "height": 45,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 45,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 22.5,
              "overflow": "hidden",
              "width": 45,
            }
          }
        >
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "height": 45,
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "rotate": "-165deg",
                  },
                ],
                "width": 45,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 22.5,
                  "overflow": "hidden",
                  "width": 45,
                }
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "borderColor": "#6200ee",
                    "borderRadius": 22.5,
                    "borderWidth": 4.5,
                    "height": 45,
                    "width": 45,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "height": 45,
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
            "width": 45,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 22.5,
              "overflow": "hidden",
              "top": 22.5,
              "width": 45,
            }
          }
        >
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "height": 45,
                "transform": Array [
                  Object {
                    "translateY": -22.5,
                  },
                  Object {
                    "rotate": "345deg",
                  },
                ],
                "width": 45,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 22.5,
                  "overflow": "hidden",
                  "width": 45,
                }
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "borderColor": "#6200ee",
                    "borderRadius": 22.5,
                    "borderWidth": 4.5,
                    "height": 45,
                    "width": 45,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
